### DISCUSSION TITLE
Editing text in the PDS files

### DISCUSSION THREAD

**Discussion by GerMFHub** [ansible-collections/ibm_zos_core#661] [Q&A]
Is there a way to with zos_core collection to replace text in the PDS file? I only saw options to add text before or after matching regex.

**Comment by ddimatos** [comment]
You should be able to use `[zos_lineinfile](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/modules/zos_lineinfile.html)` to augment text/data in a PDS/E , SEQ , MEMBER and USS file. You can also use [zos_block_infile](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/modules/zos_blockinfile.html) but it will leave managed markers behind so that the module can later react accordingly. If you don't care for the managed tags and do want to edit a block, you can always to back after `zos_blockinfile` and remove the markers with `zos_lineinfile`.

I would start with `zos_lineinfile`, optionally if you are familiar with z/OS programs you can invoke them with `zos_mvs_raw` to make edits, but it sounds like `zos_lineinfile` will work for you. 

Doc snippet of interest for zos_lineinfile:
> regexp
>     **For state=absent, the pattern of the line(s) to remove.**

Examples in doc:
```
- name: Remove all comments in the USS file
  zos_lineinfile:
    src: /tmp/src/somefile
    state: absent
    regexp: '^#'
```

**Comment by ddimatos** [comment]
Reopen if needed, closing as its past 30 days.
